### DISCUSSION TITLE
`github.action_repository` and `github.action_ref` are empty in `run` for composite actions

### DISCUSSION THREAD

**Discussion by jsoref** [community/community#49245] [Actions] [answered]
### Select Topic Area

Bug

### Body

https://docs.github.com/en/actions/learn-github-actions/contexts#github-context says:

Property name | Type | Description
-- | -- | --
github.action_repository | string | For a step executing an action, this is the owner and repository name of the action. For example, actions/checkout.
github.action_ref | string | For a step executing an action, this is the ref of the action being executed. For example, v2.

For each of the following links, expand:
> To accept ✔️ these unrecognized words as correct and remove the previously acknowledged and now absent words, run the following commands

... and examine the `curl` urls, which I've reproduced following the urls for convenience...

https://github.com/check-spelling/gotosocial/actions/runs/4336916865/attempts/1#summary-11774495078
> curl -s -S -L 'https://raw.githubusercontent.com/check-spelling/check-spelling/prerelease/apply.pl' |
perl - 'https://github.com/check-spelling/gotosocial/actions/runs/4336916865/attempts/1'

https://github.com/check-spelling/gotosocial/actions/runs/4336916865/attempts/4#summary-11774894382
> curl -s -S -L 'https://raw.githubusercontent.com/check-spelling/gotosocial/spell-check-with-spelling/apply.pl' |
perl - 'https://github.com/check-spelling/gotosocial/actions/runs/4336916865/attempts/4'

https://github.com/check-spelling/gotosocial/actions/runs/4337164095#summary-11774988109
> curl -s -S -L 'https://raw.githubusercontent.com/check-spelling/gotosocial/spell-check-with-spelling/apply.pl' |
perl - 'https://github.com/check-spelling/gotosocial/actions/runs/4337164095/attempts/1'

The expected results is that `gotosocial/spell-check-with-spelling` should not be present in the curl url (instead, it should say `check-spelling/prerelease` just as the first one does).

For my reference:
https://support.github.com/ticket/personal/0/2042233

**Comment by ChristopherHX** [comment, marked as answer]
Described in another way `github.action_repository` and `github.action_ref` are empty (**broken**) in expressions within the `run` step script definition of a composite action.

Using it in the `env` field of a run step will behave correctly. 

The change between these commits triggered an actions/runner bug
https://github.com/check-spelling/check-spelling/compare/2bb4ea78df51ab5ded5612c08586e6ed59d1f032..50ca43769650975d2dee92e29fa9cbbe30d0247b#diff-1243c5424efaaa19bd8e813c5e6f6da46316e63761421b3e5f5c8ced9a36e6b6R472

E.g. this would work, otherwise wait for GitHub Support to fix actions/runner.
```yaml
    - name: shim-path
      shell: bash
      run: |
        THIS_ACTION_PATH=$(perl -pe 's#/\./#/#g; s#//+#/#g; s#/$##g' <<< "$RAW_ACTION_PATH")
        (
          echo "THIS_ACTION_PATH=$THIS_ACTION_PATH"
          echo "PATH=$THIS_ACTION_PATH/wrappers:$PATH"
          echo "TASK=${{ inputs.custom_task || inputs.task }}"
          echo "DO_CHECKOUT=${{ inputs.checkout && inputs.checkout != 'false' && !(inputs.custom_task || inputs.task) }}"
          echo "FETCH_DEPTH=${{ !contains(inputs.check_commit_messages || inputs.check-commit-messages, 'commits') && '1' || '0' }}"
          echo "USE_SARIF=${{ inputs.use_sarif || inputs.use-sarif }}"
          echo "ALTERNATE_ENGINE=${{ (inputs.alternate_engine || inputs.alternate-engine) || '' }}"
          echo "GH_ACTION_REPOSITORY=$GH_ACTION_REPOSITORY"
          echo "GH_ACTION_REF=$GH_ACTION_REF"
        ) >> "$GITHUB_ENV"
      env:
        GH_ACTION_REPOSITORY: ${{ github.action_repository || github.repository }}
        GH_ACTION_REF: ${{ github.action_ref || github.ref_name }}
        RAW_ACTION_PATH: ${{ github.action_path }} # Your action had a script injection problem and container path translation were broken due to another bug of actions/runner
```

**Reply by jsoref** [reply]
cross filed to https://github.com/actions/runner/issues/2473

**Reply by DMkhanyt** [reply]
https://creativecommons.org/publicdomain/zero/1.0/

**Reply by DMkhanyt** [reply]
https://creativecommons.org/publicdomain/zero/1.0/
